ra/SAPHana

```diff
@@ -1236,51 +1236,47 @@ function check_for_primary() {
     # DONE: Limit the runtime of hdbnsutil and use getParameter.py as fallback
     # DONE: PRIO2: Maybe we need to use a fallback interface when hdbnsutil does not answer properly -> lookup in config files?
     # TODO:        This might also solve some problems when we could not figure-out the local or remote site name (site_name,site_id from global.ini)
-    local chkMethod=""
-    local ini_mode=""
+    local chkMethod=''
+    local node_status=''
+    local node_full_status=''
     for chkMethod in  hU hU hU gP; do
-       case "$chkMethod" in
-           gP )
-                # fallback for 'hdbnsutil' failing 3 times.
-                local gpKeys=""
+        case "$chkMethod" in
+            gP ) # call getParameter (gP) - fallback for 'hdbnsutil' failing 3 times
+                local gpKeys=''
                 gpKeys=$(echo --key=global.ini/system_replication/{actual_mode,mode})
                 node_full_status=$(HANA_CALL --timeout "$HANA_CALL_TIMEOUT" --cmd "python getParameter.py $gpKeys --sapcontrol=1" 2>&1 | awk -F/ 'BEGIN {out=0} /^SAPCONTROL-OK: <begin>/ { out=1 } /^SAPCONTROL-OK: <end>/ { out=0 } /=/ {if (out==1) {print $3} }')
-                # first try to get the value of 'actual_mode' from the global.ini
-                ini_mode=$(echo "$node_full_status" | awk -F= '$1=="actual_mode" {print $2}')
                 # if 'actual_mode' is not available, fallback to 'mode'
-                if [ -z "$ini_mode" ]; then
-                    ini_mode=$(echo "$node_full_status" | awk -F= '$1=="mode" {print $2}')
-                fi
-                node_status="$ini_mode"
+                [[ "$node_full_status" =~ (actual_mode|mode)=([^$'\n']+) ]] && node_status=${BASH_REMATCH[2]}
```

Comment:
if the order of the output lines from 'getParameter.py' will change in the future and the line 'global.ini/system_replication/mode=' will be the first one and 'global.ini/system_replication/actual_mode=' will be occur later, the original implementation will still work as expected. The new syntax will print the value from 'mode' as result instead of 'actual_mode'.
Don't know if we need to expect such a change of 'getParameter.py', but as we can not control such changes, I will stay with the original implementation to be on the  safe side.

Reply:
the order of the ouptut lines is specified by the parameters for getParameter.py

{mode,actual_mode} vs {actual_mode,mode} 
```
python getParameter.py --key=global.ini/system_replication/{mode,actual_mode} --sapcontrol=1
[140219404445440, 0.013] >> starting getParameter (at Thu Aug 17 19:28:07 2023)
[140219404445440, 0.013] args: ()
[140219404445440, 0.013] keys: {'key': ['global.ini/system_replication/mode', 'global.ini/system_replication/actual_mode'], 'sapcontrol': '1'}
SAPCONTROL-OK: <begin>
global.ini/system_replication/mode=primary
global.ini/system_replication/actual_mode=primary
SAPCONTROL-OK: <end>
[140219404445440, 0.016] << ending getParameter, rc = 0 (RC_TEST_OK), after 0.003 secs

python getParameter.py --key=global.ini/system_replication/{actual_mode,mode} --sapcontrol=1
[140435146196736, 0.014] >> starting getParameter (at Thu Aug 17 19:28:21 2023)
[140435146196736, 0.014] args: ()
[140435146196736, 0.014] keys: {'key': ['global.ini/system_replication/actual_mode', 'global.ini/system_replication/mode'], 'sapcontrol': '1'}
SAPCONTROL-OK: <begin>
global.ini/system_replication/actual_mode=primary
global.ini/system_replication/mode=primary
SAPCONTROL-OK: <end>
```

Reply:
> the order of the ouptut lines is specified by the parameters for getParameter.py
> 
> {mode,actual_mode} vs {actual_mode,mode}

That was my first idea, but on my test system I got the same order irrespective of the order in the 'key'

```
python getParameter.py --key=global.ini/system_replication/{mode,actual_mode} --sapcontrol=1
[140190982116288, 0.001] >> starting getParameter (at Thu Aug 17 20:12:55 2023)
[140190982116288, 0.001] args: ()
[140190982116288, 0.001] keys: {'key': ['global.ini/system_replication/mode', 'global.ini/system_replication/actual_mode'], 'sapcontrol': '1'}
SAPCONTROL-OK: <begin>
global.ini/system_replication/actual_mode=primary
global.ini/system_replication/mode=primary
SAPCONTROL-OK: <end>
[140190982116288, 0.003] << ending getParameter, rc = 0 (RC_TEST_OK), after 0.001 secs
```
So this different behavior between your system and my system is one more argument to stay with the 'old' implementation as it will work on both of our systems.

Reply:
On my system (for other parameters) I always get the same order independent from the keys. But it looks like it is a reverse alphabetical order in my system. This again shows we should not trust the order of these lines. I would like to try to get the values separately ($actual_mode and $mode) and then using $actual_mode, if available and $mode, if not (hope I did it the right way) ;-).

Reply:
yes, that was my intention behind "stay with the 'old' implementation". We can exchange the awk usage with bash syntax, but we should extract the value in 2 steps to guarantee the important order.

@PeterPitterling  If you don't mind I will close this PR and #158 too, but will include the change from 'awk' to 'pure bash' in my currently ongoing fix for another bug. So you do not need to put extra effort in changing these PRs or add a new one. Hope that's ok for you.

Reply:
agree for this PR .. #158 contains a bit more then just the awk replacement.

I assume, you're going to include the below changes as well 
sapcontrol, awk for node_full_status, move sleep, i++

Reply:
> I assume, you're going to include the below changes as well sapcontrol, awk for node_full_status, move sleep, i++

yes, sure. I will add all additional changes of PR158 and 159 too.

Reply:
@PeterPitterling  - see https://github.com/SUSE/SAPHanaSR/pull/196 for the changes.